Mathlib/Analysis/Analytic/IsolatedZeros.lean

```diff
@@ -240,6 +240,44 @@ lemma apply_eq_zero_of_order_toNat_ne_zero (hf : AnalyticAt 𝕜 f z₀) :
   simp [hf.order_eq_zero_iff]
   tauto
 
+/- Helper lemma for `AnalyticAt.order_mul` -/
+lemma order_eq_top_of_order_eq_top_mul_analytic {f g : 𝕜 → 𝕜} (hf : AnalyticAt 𝕜 f z₀)
+  (hg : AnalyticAt 𝕜 g z₀) (h'f : hf.order = ⊤) :
+    (hf.mul hg).order = ⊤ := by
+  rw [AnalyticAt.order_eq_top_iff, eventually_nhds_iff] at *
+  obtain ⟨t, h₁t, h₂t, h₃t⟩ := h'f
+  exact ⟨t, fun y hy ↦ (by rw [h₁t y hy]; ring), h₂t, h₃t⟩
+
+/-- The order is additive when multiplying analytic functions. -/
+theorem order_mul {f g : 𝕜 → 𝕜} (hf : AnalyticAt 𝕜 f z₀) (hg : AnalyticAt 𝕜 g z₀) :
+    (hf.mul hg).order = hf.order + hg.order := by
+  -- Trivial cases: one of the functions vanishes around z₀
+  by_cases h₂f : hf.order = ⊤
+  · simp [hf.order_eq_top_of_order_eq_top_mul_analytic hg h₂f, h₂f]
+  by_cases h₂g : hg.order = ⊤
+  · have : (fun x ↦ f x * g x) = (fun x ↦ g x * f x) := by simp_rw [mul_comm]
+    simp [this, hg.order_eq_top_of_order_eq_top_mul_analytic hf h₂g, h₂g]
+
+  -- Non-trivial case: both functions do not vanish around z₀
+  obtain ⟨g₁, h₁g₁, h₂g₁, h₃g₁⟩ := hf.order_neq_top_iff.1 h₂f
+  obtain ⟨g₂, h₁g₂, h₂g₂, h₃g₂⟩ := hg.order_neq_top_iff.1 h₂g
+  rw [← ENat.coe_toNat h₂f, ← ENat.coe_toNat h₂g, ← ENat.coe_add, (hf.mul hg).order_eq_nat_iff]
+  use g₁ * g₂, by exact h₁g₁.mul h₁g₂
+  constructor
+  · simp
+    tauto
+  · obtain ⟨t, h₁t, h₂t, h₃t⟩ := eventually_nhds_iff.1 h₃g₁
+    obtain ⟨s, h₁s, h₂s, h₃s⟩ := eventually_nhds_iff.1 h₃g₂
+    exact eventually_nhds_iff.2
+      ⟨t ∩ s, fun y hy ↦ (by rw [h₁t y hy.1, h₁s y hy.2]; simp; ring), h₂t.inter h₂s, h₃t, h₃s⟩
+
+/-- The order multiplies by `n` when taking an analytic function to its `n`th power. -/
+theorem order_pow {f : 𝕜 → 𝕜} (hf : AnalyticAt 𝕜 f z₀) {n : ℕ} :
+    (hf.pow n).order = n * hf.order := by
+  induction' n with n hn
```

Comment:
For new code, my personal preference is `induction` without prime

Reply:
@Ruben-VandeVelde Thanks for your suggestion that I would like to implement. I am a beginner, however, and I am struggling to make the tactic `induction` work -- I learned from `Mathematics in Lean` that only ever uses `induction'`. Can you please help or point me to a reference? Thank you so much! -- Best, Stefan.

Reply:
@Ruben-VandeVelde Disregard my previous mail. Meanwhile I found [this](https://seasawher.github.io/mathlib4-help/tactics/#induction) and switched to the `induction` tactic as you suggested.